runtime/arm.S

```diff
@@ -99,11 +99,27 @@ alloc_limit     .req    r11
 #define CFI_OFFSET(r,n)
 #endif
 
-/* Allocation functions and GC interface */
+#if defined(FUNCTION_SECTIONS)
+#define TEXT_SECTION(name) .section .text.##name,"ax",%progbits
+#else
+#define TEXT_SECTION(name)
+#endif
+
+#if defined(FUNCTION_SECTIONS)
+        TEXT_SECTION(caml_hot__code_begin)
+        .globl  caml_hot__code_begin
+caml_hot__code_begin:
 
+        TEXT_SECTION(caml_hot__code_end)
+        .globl  caml_hot__code_end
+caml_hot__code_end:
+#endif
+
+/* Allocation functions and GC interface */
         .globl  caml_system__code_begin
 caml_system__code_begin:
 
+        TEXT_SECTION(caml_call_gc)
         .align  2
         .globl  caml_call_gc
```

Comment:
The duplication of the name in the macro and `.globl` seems dangerous. Maybe a single macro for declaring both would be better.

Reply:
The name appeared several times, unfortunately, even before this patch.  I haven't fixed it before as I wanted to keep the changes in this PR to minimum and only relevant to function sections.

I have just pushed an update that cleans it up a bit.  It replaces all the uses of names at the start of functions with a macro, similarly to amd64 backend.  There is another appearance at the end of functions, but I don't know how to fix it nicely in the same macro.

Running it on precheck now.